salt/api.sls

```diff
@@ -1,16 +1,47 @@
+#!jinja|yaml
+
 {% from "salt/map.jinja" import salt_settings with context %}
 
 include:
   - salt.master
+  - pip.extensions
+
+{%- set cfg_salt = pillar.get('salt', {}) %}
+{%- set cfg_master = cfg_salt.get('master', {}) %}
 
-salt-api:
-{% if salt_settings.install_packages %}
+salt_api_install:
   pkg.installed:
-    - name: {{ salt_settings.salt_api }}
-{% endif %}
+    - name: {{ salt_settings['salt_api'] }}
   service.running:
-    - name: {{ salt_settings.api_service }}
+    - name: {{ salt_settings.get('api_service', 'salt-api') }}
     - require:
-      - service: {{ salt_settings.master_service }}
+{%- if 'rest_cherrypy' in cfg_master %}
+      - pip: salt_api_cherrypy
+{% elif 'rest_tornado' in cfg_master %}
+      - pip: salt_api_tornado
+{% endif %}
     - watch:
       - pkg: salt-master
+      - file: salt-master
+
+{%- if 'rest_cherrypy' in cfg_master %}
+salt_api_cherrypy:
+  pkg.purged:
+    - name: {{ salt_settings['python-cherrypy'] }}
```

Comment:
Installing from pip vs from package should be completely optional. Some people have perfectly valid versions of cherrypy in their package manager.